app/src/androidTest/java/fr/free/nrw/commons/AboutActivityTest.kt

```diff
@@ -0,0 +1,116 @@
+import androidx.test.core.app.ApplicationProvider.getApplicationContext
+import androidx.test.espresso.Espresso
+import androidx.test.espresso.action.ViewActions
+import androidx.test.espresso.assertion.ViewAssertions
+import androidx.test.espresso.intent.Intents
+import androidx.test.espresso.intent.matcher.IntentMatchers
+import androidx.test.espresso.matcher.ViewMatchers
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import androidx.test.rule.ActivityTestRule
+import androidx.test.runner.AndroidJUnit4
+import fr.free.nrw.commons.utils.ConfigUtils
+import org.hamcrest.CoreMatchers
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class AboutActivityTest {
+    @get:Rule
+    var activityRule: ActivityTestRule<*> = ActivityTestRule(AboutActivity::class.java)
+
+    @Before
+    fun setup() {
+        Intents.init()
+        UITestHelper.skipWelcome()
+        UITestHelper.skipLogin()
+        Intents.intending(CoreMatchers.not(IntentMatchers.isInternal()))
+                .respondWith(Instrumentation.ActivityResult(Activity.RESULT_OK, null))
+    }
+
+    @Test
+    fun testBuildNumber() {
+        Espresso.onView(ViewMatchers.withId(R.id.about_version))
+                .check(ViewAssertions.matches(withText(ConfigUtils.getVersionNameWithSha(getApplicationContext()))))
+    }
+
+    @Test
+    fun testLaunchWebsite() {
+        Espresso.onView(ViewMatchers.withId(R.id.website_launch_icon)).perform(ViewActions.click())
+        Intents.intended(CoreMatchers.allOf(IntentMatchers.hasAction(Intent.ACTION_VIEW),
+                IntentMatchers.hasData(Urls.WEBSITE_URL)))
+    }
+
+    @Test
+    fun testLaunchFacebook() {
+        Espresso.onView(ViewMatchers.withId(R.id.facebook_launch_icon)).perform(ViewActions.click())
+        try {
+            Intents.intended(CoreMatchers.allOf(IntentMatchers.hasAction(Intent.ACTION_VIEW),
+                    IntentMatchers.hasPackage(Urls.FACEBOOK_PACKAGE_NAME)))
+        } catch (e: Exception) {
+            Intents.intended(CoreMatchers.allOf(IntentMatchers.hasAction(Intent.ACTION_VIEW),
+                    IntentMatchers.hasData(Urls.FACEBOOK_WEB_URL)))
+        }
+    }
+
+    @Test
+    fun testLaunchGithub() {
+        Espresso.onView(ViewMatchers.withId(R.id.github_launch_icon)).perform(ViewActions.click())
+        Intents.intended(CoreMatchers.allOf(IntentMatchers.hasAction(Intent.ACTION_VIEW),
+                IntentMatchers.hasData(Urls.GITHUB_REPO_URL)))
+    }
+
+    @Test
+    fun testLaunchRateUs() {
+        val appPackageName = BuildConfig::class.java.getPackage().name
+        Espresso.onView(ViewMatchers.withId(R.id.about_rate_us)).perform(ViewActions.click())
+        try {
+            Intents.intended(CoreMatchers.allOf(IntentMatchers.hasAction(Intent.ACTION_VIEW),
+                    IntentMatchers.toPackage(Urls.PLAY_STORE_PACKAGE_NAME)))
+        } catch (e: Exception) {
```

Comment:
generally a test should not be catching exceptions

Reply:
@macgills I used the try-except to handle the different cases just as it was used in the AboutActivity but I believe there should be a better way but I can't find one.

Reply:
This is how you can check for an installed application, [link](https://stackoverflow.com/questions/18752202/check-if-application-is-installed-android) but a test shouldn't perform this.

A test should exist for each flow, ideally you can say in the test beforehand "report that the playstore isn't installed". The lack of abstraction in the source will make this difficult to achieve

Reply:
It is ugly as sin but I am pretty sure you can just or EspressoMatchers so do that and avoid the try catch

Reply:
@macgills Should I update it to AboutActivity.java also?

Reply:
I'd say out of scope, leave it for now

Reply:
@macgills Can I use anyOf like this
```
val appPackageName = BuildConfig::class.java.getPackage().name
        Espresso.onView(ViewMatchers.withId(R.id.about_rate_us)).perform(ViewActions.click())
        Intents.intended(IntentMatchers.hasAction(Intent.ACTION_VIEW))
        Intents.intended(CoreMatchers.anyOf(IntentMatchers.toPackage(Urls.PLAY_STORE_PACKAGE_NAME),
                IntentMatchers.hasData("${Urls.PLAY_STORE_URL_PREFIX}$appPackageName")))
```
Also, regarding @maskaravivek suggestion, I am unable to extract the packageName for play store.

Reply:
Don't get the package name like that `InstrumentationRegistry.getInstrumentation().getTargetContext()` to get the context then get the package name from the context

Reply:
@macgills I have fixed the appPackageName

But I am still not able to get the PLAY STORE package name.

Reply:
That is fine to be hardcoded, you just shouldn't hardcode our package name because we can retrieve that